src/V3RandomizeMethod.cpp

```diff
@@ -0,0 +1,171 @@
+            auto* fvarp = VN_CAST(funcp->fvarp(), Var);
+            funcp->addStmtsp(new AstAssign(
+                nodep->fileline(), new AstVarRef(nodep->fileline(), fvarp, VAccess::WRITE),
+                new AstConst(nodep->fileline(), AstConst::WidthedValue(), 32, 1)));
+            auto* classp = nodep;
+            for (auto* classp = nodep; classp;
+                 classp = classp->extendsp() ? classp->extendsp()->classp() : nullptr) {
+                for (auto* memberp = classp->stmtsp(); memberp; memberp = memberp->nextp()) {
+                    auto* memberVarp = VN_CAST(memberp, Var);
+                    if (!memberVarp || !memberVarp->isRand()) continue;
+                    AstNode* stmtp = nullptr;
+                    auto* refp = new AstVarRef(nodep->fileline(), memberVarp, VAccess::WRITE);
+                    if (VN_IS(memberp->dtypep()->skipRefp(), BasicDType)) {
+                        stmtp = new AstStdRandomize(nodep->fileline(), refp);
+                    } else if (auto* classRefp = VN_CAST(memberp->dtypep(), ClassRefDType)) {
+                        auto* funcp = V3RandomizeMethod::declareIn(classRefp->classp());
+                        auto* callp
+                            = new AstMethodCall(nodep->fileline(), refp, "randomize", nullptr);
+                        callp->taskp(funcp);
+                        callp->dtypeFrom(funcp);
+                        stmtp = callp;
+                    } else {
+                        delete refp;
+                        memberp->v3warn(E_UNSUPPORTED,
+                                        "Unsupported: random member variables with type "
+                                            << memberp->dtypep()->prettyDTypeNameQ());
+                    }
+                    if (stmtp) {
+                        // Although randomize returns int, we know it is 0/1 so can use faster
+                        // AstAnd vs AstLogAnd.
+                        stmtp = new AstAnd(nodep->fileline(),
+                                           new AstVarRef(nodep->fileline(), fvarp, VAccess::READ),
+                                           stmtp);
+                        auto* assignp = new AstAssign(
+                            nodep->fileline(),
+                            new AstVarRef(nodep->fileline(), fvarp, VAccess::WRITE), stmtp);
+                        funcp->addStmtsp(assignp);
+                    }
+                }
+            }
+            nodep->user1(false);
+        }
+    }
+};
+
+//######################################################################
+// Randomize method class functions
+
+void V3RandomizeMethod::defineIfNeeded(AstNetlist* nodep) {
+    UINFO(2, __FUNCTION__ << ": " << endl);
+    {
+        RandomizeMethodMarkVisitor prepVisitor;
+        prepVisitor.iterate(nodep);
+        prepVisitor.markAllDerived();
+        RandomizeMethodVisitor visitor;
+        visitor.iterate(nodep);
```

Comment:
```suggestion
        RandomizeMethodVisitor visitor{nodep};
```

And have constructor iterate (see some other Visitors)